### PR TITLE
test: File Readers and Text Processing Utils (Track A)

### LINKED ISSUE #379: [Testing] Story 2: File Readers Test Coverage
## Story 2: File Readers Test Coverage

**Epic:** Testing & QA (#6)
**Effort:** 5-6 hours
**Expected coverage gain:** +6-8%

## Scope

- Create/extend `tests/utils/test_file_readers.py` for text-based readers
- Readers: `read_text_file`, `read_docx_file`, `read_pdf_file`, `read_spreadsheet_file`, `read_presentation_file`, `read_ebook_file`, `read_csv_file`
- Add minimal fixture files (`.docx`, `.pdf`, `.xlsx`, `.csv`)
- Test `_check_file_size()`, error paths (FileReadError, FileTooLargeError)
- Test `*_AVAILABLE=True` and `*_AVAILABLE=False` paths
- Skip audio/video/image readers (deferred)

## Files

- `tests/utils/test_file_readers.py`
- `tests/fixtures/`
- `src/file_organizer/utils/file_readers.py`

## Acceptance Criteria

- [ ] Tests for all 7 text-based file readers
- [ ] Error path coverage (FileReadError, FileTooLargeError)
- [ ] Optional dependency availability paths tested
- [ ] Minimal fixture files committed
- [ ] All new tests passing

### LINKED ISSUE #381: [Testing] Story 4: Text Processing Utils Test Coverage
## Story 4: Text Processing Utils Test Coverage

**Epic:** Testing & QA (#6)
**Effort:** 2-3 hours
**Expected coverage gain:** +1-2%

## Scope

- Create `tests/utils/test_text_processing.py`
- Pure function unit tests — no mocking needed
- Text normalization, extraction, cleaning utilities

## Files

- `tests/utils/test_text_processing.py`
- `src/file_organizer/utils/text_processing.py`

## Acceptance Criteria

- [ ] All public functions in text_processing.py tested
- [ ] Edge cases: empty strings, unicode, special characters
- [ ] All new tests passing

### PR DESCRIPTION
**Pull request by curdriceaurora**
Resolves #379 and Resolves #381. Adds high quality unit testing coverage to file readers and text processing.

### CODE REVIEW

File: tests/utils/test_file_readers.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    @patch("file_organizer.utils.file_readers.EBOOKLIB_AVAILABLE", False)
+    def test_read_ebook_not_installed(self):
+        with pytest.raises(ImportError, match="ebooklib is not installed"):
+            read_ebook_file("test.epub")
+
+    @patch("file_organizer.utils.file_readers.EBOOKLIB_AVAILABLE", True)
+    def test_read_ebook_unsupported_format(self, tmp_path):
+        test_file = tmp_path / "test.mobi"
+        test_file.touch()
+        with pytest.raises(ValueError, match="Unsupported ebook format"):
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/utils/test_file_readers.py]: #422 422
